### ISSUE TITLE
publishToConfluence removes all linebreaks from code

### ISSUE THREAD

**Issue by tkleiber** [docToolchain/docToolchain#1253]
**Describe the bug**
When I have multiline code snippets, publishToConfluence removes all linebreaks from code, the html file does have have all of it.

**To Reproduce**
1. Create a file with embedded or include code with at least 2 lines, for example
```
* Update Jenkins Version
** Anmeldung als root
** Stoppen/Update/Starten Jenkins
+
[source,bash]
----
systemctl stop jenkins
zypper update jenkins
systemctl start jenkins
----
```
3. Run publishToConfluence

**Expected behavior**
Confluence should show all lines of code separately instead of 1 concatenated line.

**Screenshots**
generated html file:
![image](https://github.com/docToolchain/docToolchain/assets/2185441/4575813a-53dc-4eca-a25c-8d43e640d303)
exported to confluence:
![image](https://github.com/docToolchain/docToolchain/assets/2185441/7306050d-8db0-4ae9-8d4c-9734b45aba89)

**Configuration**
 - docToolchain version: 3.0.2
 - OS: Linux SLES 15 SP 4
 - Browser: Edge, Chrome
 - Relevant SDK Versions: Temurin jdk-17.0.6+10

**Comment by PacoVK** [comment]
Thanks for reporting this. Unfortunately I cannot reproduce it. Are you on Confluence Cloud or server? If Server which version do you run?

**Comment by tkleiber** [comment]
Confluence Server 8.22.6

**Comment by PacoVK** [comment]
@tkleiber are you sure, i can only find Jira Software releases that match this version. https://confluence.atlassian.com/display/DOC/Confluence+8.6+Release+Notes

**Comment by tkleiber** [comment]
Sorry, my fault. This was the JIRA version.
Confluence is 7.19.12.
Will be updated in the next days to 7.19.16

**Comment by PacoVK** [comment]
@tkleiber did you upgrade your confluence meanwhile, and does the issue still persist?

**Comment by tkleiber** [comment]
Tested both on Confluence Version 7.19.16 (Server) and 8.5.3 (Data Center).

Issue is reproducible on docToolchain 3.0.2 and 3.1.2.
Issue is not reproducible on docToolchain 2.1.0.

**Comment by PacoVK** [comment]
Thanks I keep trying to investigate, although it's hard without access to a Server instance. But actually you provided good hints 👍

**Comment by stehlih** [comment]
We have the same problem on our on-premise Confluence server 8.6.1

We are using doctoolchain 3.1.2.

If subpagesForSections = 1 all source code blocks are published with the right line breaks.

After switch to subpagesForSections = 2 all source code blocks are now OneLiner (all line breaks are removed),

Going back to  subpagesForSections = 1 restores the line breaks (and we must delete all the ghost subpages).

**Comment by tkleiber** [comment]
We need subpagesForSections = 2. Maybe the issue is related to the changes for my issue #402?

**Comment by PacoVK** [comment]
Awesome thanks the last hint was the missing part and I could now reproduce and locate the error in rewriteCodeBlocks. However the bug seems not to be trivial (after first investigations) and I need some time to come up with a fix...

**Comment by PacoVK** [comment]
Here we go, thanks to your input @tkleiber and @stehlih  i could finally find the issue, reproduce and resolve it. 
Will be fixed as soon as #1306  gets live :)
